src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/repository/BackupsRepository.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * The MIT License
+ * Copyright (c) 2018 Estonian Information System Authority (RIA),
+ * Nordic Institute for Interoperability Solutions (NIIS), Population Register Centre (VRK)
+ * Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.niis.xroad.restapi.repository;
+
+import ee.ria.xroad.common.SystemProperties;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Repository;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.FileTime;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Backups repository
+ */
+@Slf4j
+@Repository
+public class BackupsRepository {
+
+    private static final String CONFIGURATION_BACKUP_PATH = SystemProperties.getConfBackupPath();
+    private static final String BACKUP_FILE_EXTENSION = ".tar";
+    // Set maximum number of levels of directories to visit, subdirectories are excluded
+    private static final int DIR_MAX_DEPTH = 1;
+
+    /**
+     * Read backup files from configuration backup path
+     * @return
+     */
+    public List<File> getBackupFiles() {
+        try (Stream<Path> walk = Files.walk(Paths.get(CONFIGURATION_BACKUP_PATH), DIR_MAX_DEPTH)) {
+            return walk.map(Path::toString)
+                    .filter(f -> f.endsWith(BACKUP_FILE_EXTENSION))
```

Comment:
Old implementation is more strict in filtering:
- https://github.com/nordic-institute/X-Road/blob/09f42ff76da5aa42da09d58fd287246e25e8bc36/src/common-ui/lib/common-ui/backup_utils.rb#L51-L52
- https://github.com/nordic-institute/X-Road/blob/09f42ff76da5aa42da09d58fd287246e25e8bc36/src/common-ui/lib/common-ui/io_utils.rb#L139-L141

Should we implement the same logic? I think this implementation would list more backups than the old UI (e.g. backup files starting with a dot)

Reply:
The new UI/API implements now the same validation rules that the old UI does. Subdirectories are excluded using `DIR_MAX_DEPTH = 1` and valid file names are limited to `"^(?!\\.)[\\w\\.\\-]+\\.tar$"` pattern.

Reply:
Very nice! Documenting what the regexp does in https://github.com/nordic-institute/X-Road/blob/7187427ed43ff2f7334fcc221c0df08a4359e57d/src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/repository/BackupRepository.java#L53-L54 is also really great, since regexps are difficult to read and adding a few lines of comments helps reviewer / modifier a lot.